data-serving/reusable-data-service/tests/test_case_end_to_end.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+import mongomock
+import pymongo
+
+from reusable_data_service import app, set_up_controllers
+
+
+@pytest.fixture
+def client_with_patched_mongo(monkeypatch):
+    # configure controllers
+    monkeypatch.setenv("DATA_STORAGE_BACKEND", "mongodb")
+    monkeypatch.setenv(
+        "MONGO_CONNECTION_STRING", "mongodb://localhost/27017"
+    )  # will be unused
+    monkeypatch.setenv("MONGO_DB", "outbreak")
+    monkeypatch.setenv("MONGO_CASE_COLLECTION", "cases")
+    db = mongomock.MongoClient()
+
+    def fake_mongo(connection_string):
+        return db
+
+    monkeypatch.setattr("pymongo.MongoClient", fake_mongo)
+    set_up_controllers()
+    app.config["TESTING"] = True
+    client = app.test_client()
+    yield client
+
+
+def test_get_case_with_known_id(client_with_patched_mongo):
+    # insert a case
+    db = pymongo.MongoClient("mongodb://localhost/27017")
```

Comment:
localhost:27017

Reply:
In practice not a problem because the connection string is ignored in tests but I'll definitely correct it so as not to cause any further confusion :).